components/Filter/index.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import TechnologyDropdown from '@/components/TechnologyDropdown';
+import { Filters, FilterType } from '@/types/schema';
+import { DropIcon } from '../../assets/Dropdown-Icons/icons';
+import {
+  FilterBackgroundStyles,
+  FilterButtonStyles,
+  IconStyle,
+} from './styles';
+
+interface FilterProps {
+  filter: FilterType;
+  isActive: boolean;
+  selectedFilters: Filters;
+  onTechnologyChange: (options: string[]) => void;
+  handleButtonClick: (filter: FilterType) => void;
+}
+
+export default function Filter({
+  filter,
+  isActive,
+  selectedFilters,
+  onTechnologyChange,
+  handleButtonClick,
+}: FilterProps) {
+  return (
+    <FilterBackgroundStyles isActive={isActive}>
+      {isActive ? (
+        filter.id === 'technology' ? (
+          <TechnologyDropdown
+            selectedTechnologies={selectedFilters.technology}
+            setSelectedTechnologies={onTechnologyChange}
+            handleButtonClick={handleButtonClick}
+            icon={filter.icon}
+            label={filter.label}
+            currFilter={filter}
+          />
+        ) : // Add other filter dropdown components here
+        null
+      ) : (
+        <FilterButtonStyles onClick={() => handleButtonClick(filter)}>
+          <IconStyle>{filter.icon}</IconStyle>
```

Comment:
the icon is wrong. It should be the energy icon
<img width="197" alt="Screenshot 2024-11-12 at 5 23 46 PM" src="https://github.com/user-attachments/assets/ddce9068-602d-4e15-84f9-c545a75ddcdb">